File: Array/easy/finding_max_and_min_single_scan.cpp
//
// Created by santosh on 12/1/25.
//


#include<bits/stdc++.h>
using namespace std;

pair<int, int> getMinAndMaxInAnArray(const vector<int>& arr)
{
    auto mn = INT_MAX;
    auto mx = INT_MIN;
    for (int i = 0; i < arr.size(); i++)
    {
        if (arr[i] > mx)
        {
            mx = arr[i];
        }
        else if (arr[i] < mn)
        {
            mn = arr[i];
        }
    }
    return {mn, mx};
};

int main()
{
    auto arr = {6, 3, 8, 10, 16, 7, 5, 2, 9, 14};
    auto [mn, mx] = getMinAndMaxInAnArray(arr);
    cout << "MINIMUM" <<  mn << endl;
    cout << "MAXIMUM" << mx << endl;
}

File: Array/easy/finding_max_and_min_single_scan.py
# Find min and max in a single scan
# Created by santosh

def get_min_and_max_in_an_array(arr: list[int]) -> tuple[int, int]:
    mn = float('inf')
    mx = float('-inf')
    for val in arr:
        if val > mx:
            mx = val
        elif val < mn:
            mn = val
    return mn, mx


if __name__ == "__main__":
    arr = [6, 3, 8, 10, 16, 7, 5, 2, 9, 14]
    mn, mx = get_min_and_max_in_an_array(arr)
    print("MINIMUM", mn)
    print("MAXIMUM", mx)
